config/initializers/vault.rb

```diff
@@ -59,44 +62,43 @@ def initialize
           writer = Vault::Client.new(DEFAULT_CLIENT_OPTIONS.merge(client_cert_options))
           writer.token = VaultClient.auth_token(vault_server["vault_address"])
         end
-        writer.address = vault_server["vault_address"]
-        writer.ssl_ca_cert = vault_server["ca_cert"] if vault_server["ca_cert"]
-        writer
+        @vaults[instance].address = vault_server["vault_address"]
+        @vaults[instance].ssl_ca_cert = vault_server["ca_cert"] if vault_server["ca_cert"]
       end
-      @reader = @writers.first
     end
 
+    # normally we'll only get a single instance back.  if it's global, just read it once
     def read(key)
-      @reader.logical.read(key)
+      vault_instances(key).first.logical.read(key)
     end
 
     def list(path)
-      @reader.logical.list(path)
+      @vaults.flat_map { |vault| vault.pop.logical.list(path) }
     end
 
     # make darn sure on deletes and writes that we try a couple of times.
     # not going to catch any other excpeptions as we want this to blow up
     # if anything fails
     def write(key, data)
-      @writers.each do |vault_server|
-        Vault.with_retries(Vault::HTTPConnectionError, attempts: 5) do
-          vault_server.logical.write(key, data)
-        end
+      Vault.with_retries(Vault::HTTPConnectionError, attempts: 5) do
+        vault_instances(key).each { |v| v.logical.write(key, data) }
       end
     end
 
     def delete(key)
-      @writers.each do |vault_server|
-        Vault.with_retries(Vault::HTTPConnectionError, attempts: 5) do
-          vault_server.logical.delete(key)
-        end
+      Vault.with_retries(Vault::HTTPConnectionError, attempts: 5) do
+        vault_instances(key).each { |v| v.logical.delete(key) }
       end
     end
 
-    def config_for(deploy_group_name)
-      vault_hosts.detect do |hash|
-        hash.fetch('deploy_groups', []).include? deploy_group_name
-      end
+    def self.available_instances
+      client = self.new
+      client.send(:ensure_config_exists)
```

Comment:
make the method static so we don't have to create a instance and throw it away ?